extensions/amp-story/1.0/page-advancement.js

```diff
@@ -424,20 +456,35 @@ class ManualAdvancement extends AdvancementConfig {
    */
   shouldHandleEvent_(event) {
     let shouldHandleEvent = false;
+    let interactiveElementDetected = false;
     let tagName;
 
     closest(
       dev().assertElement(event.target),
       el => {
         tagName = el.tagName.toLowerCase();
-        if (
-          tagName === 'amp-story-page-attachment' ||
-          tagName === 'amp-story-quiz'
-        ) {
+
+        if (tagName === 'amp-story-page-attachment') {
           shouldHandleEvent = false;
           return true;
         }
 
+        if (tagName == 'amp-story-quiz') {
+          shouldHandleEvent = false;
+          interactiveElementDetected = true;
+        }
+
+        // If the click came from an interactive element,
+        // check whether it's inside the interactive area
+        if (interactiveElementDetected && tagName === 'amp-story-page') {
+          const withinInteractiveArea = this.isWithinInteractiveArea_(
+            event,
+            el
+          );
+          shouldHandleEvent = !withinInteractiveArea;
+          return withinInteractiveArea;
+        }
```

Comment:
We can make this code easier: if the tap was by the right/left edge, always handle it and return. We don't need to know if it's on a quiz or a tweet, we know we want to navigate (as long as it's a descendant of `amp-story-page`).

Also, please check if the other instances of `this.isInScreenSideEdge_()` need to be kept or can be removed since you're checking it earlier in the process here.

Reply:
I moved the `isInScreenSideEdge_` check before the call to `closest` which simplifies things somewhat, but the one other usage of `isInScreenSideEdge_` can't be simplified here without refactoring quizzes tweets and other interactive elements to work the same way.

Reply:
You should be able to simplify further after you resolve the other one comment :)